datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeRecoveryFlowEventChainFactory.java

```diff
@@ -0,0 +1,68 @@
+package com.sequenceiq.datalake.flow.chain;
+
+import static com.sequenceiq.datalake.flow.delete.SdxDeleteEvent.SDX_DELETE_EVENT;
+import static com.sequenceiq.datalake.flow.detach.SdxDetachEvent.SDX_DETACH_RECOVERY_EVENT;
+import static com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent.SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+import static com.sequenceiq.datalake.flow.start.SdxStartEvent.SDX_START_EVENT;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.flow.delete.event.SdxDeleteStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.SdxStartDetachRecoveryEvent;
+import com.sequenceiq.datalake.flow.start.event.SdxStartStartEvent;
+import com.sequenceiq.flow.core.chain.FlowEventChainFactory;
+import com.sequenceiq.flow.core.chain.config.FlowTriggerEventQueue;
+
+@Component
+public class DatalakeResizeRecoveryFlowEventChainFactory implements FlowEventChainFactory<DatalakeResizeRecoveryFlowChainStartEvent> {
+    @Override
+    public String initEvent() {
+        return SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+    }
+
+    @Override
+    public FlowTriggerEventQueue createFlowTriggerEventQueue(DatalakeResizeRecoveryFlowChainStartEvent event) {
+        Queue<Selectable> flowChain = new ConcurrentLinkedQueue<>();
+
+        // Forced deletion of the resized datalake.
+        boolean firstEventAdded = false;
+        if (event.getNewCluster().getDeleted() == null) {
```

Comment:
When would this be true?

Reply:
I think you mean to ask when would this be false since it is expected that recovery is done when the resized DL has failed creation or something else.

As to when it would be false, I'm not sure. I wanted to add more flexibility to the recovery process in case we wanted to retry recovery (after it already deleted the resized DL) or if we wanted to allow recovery before the resized DL is even created.

I am open to removing this though, it's not super important.